services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -239,55 +251,64 @@ func (f *BlockPersistence) WriteNextBlock(blockPair *protocol.BlockPairContainer
 	f.blockWriter.Lock()
 	defer f.blockWriter.Unlock()
 
-	bh := blockPair.ResultsBlock.Header.BlockHeight()
-
-	currentTop := f.bhIndex.getLastBlockHeight()
-	if bh != currentTop+1 {
-		return false, currentTop, nil
+	bh := getBlockHeight(blockPair)
+	lastWrittenHeight := f.bhIndex.lastSyncedHeight
+	if _, found := f.bhIndex.fetchBlockOffset(bh); found { // block already in storage, ignores as defined in test expected
```

Comment:
this is an extra safety check to avoid writing an existing block - perhaps can be removed - I'm not sure why to optimize.
The next expected offset for writing is maintained in memory regardless of map